### PR TITLE
Heretics Balancing patch

### PR DESCRIPTION
**Pull request by Djiq**
## About The Pull Request

Replaces 'touch of madness' on ash path with fiery rebirth, an AOE 'coup de grace' style move that heals you for every person on fire that is alive near you. It finishes off people in critical condition.

Replaces Cleave with blood siphon and blood siphon with cleave. Switches their positions. Cleave is a very op spell, and im moving it up the skill tree so you cannot research it as soon as you normally could.

## Why It's Good For The Game

Cleave was easy to get, now it requires you to work more.

Touch of madness was an useless spell, i replaced it with something managable, since ash path doesnt really have many abilities, unlike rust.

## Changelog
:cl:
balance: Replaced touch of madness with a new fun late game spell
balance: Switches positions of blood siphon and cleave
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
I just hope it heals faster then fire burns you.

Number 1 rule of being set on fire by your opponent is to set them on fire as well before rolling it out.

**Comment by EdgeLordExe** [comment]
> I just hope it heals faster then fire burns you.
> 
> Number 1 rule of being set on fire by your opponent is to set them on fire as well before rolling it out.

oh right i forgot to make it also extinguish you.

File: code/modules/antagonists/eldritch_cult/eldritch_magic.dm [hunk 1]
```diff
@@ -474,3 +474,40 @@
...
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "smoke"
+
+/obj/effect/proc_holder/spell/targeted/fiery_rebirth/cast(list/targets, mob/user)
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/human_user = user
+	for(var/mob/living/carbon/target in view(7,user))
+		if(target.stat != DEAD && target.on_fire)
+			//This is essentially a death mark, use this to finish your opponent quicker.
```
Comment (Rohesie) [inline code/modules/antagonists/eldritch_cult/eldritch_magic.dm:501]:
```suggestion
		if(target.stat == DEAD || !target.on_fire)
			continue
		//This is essentially a death mark, use this to finish your opponent quicker.
```
Early continue

File: code/modules/antagonists/eldritch_cult/eldritch_magic.dm [hunk 2]
```diff
@@ -474,3 +474,40 @@
...
+			if(target.InCritical())
+				target.death()
+			target.adjustFireLoss(20)
+			new /obj/effect/temp_visual/eldritch_smoke(target.drop_location())
+			human_user.ExtinguishMob()
+			human_user.adjustBruteLoss(-10)
+			human_user.adjustFireLoss(-10)
+			human_user.adjustStaminaLoss(-10)
+			human_user.adjustToxLoss(-10)
+			human_user.adjustOxyLoss(-10)
```
Comment (Rohesie) [inline code/modules/antagonists/eldritch_cult/eldritch_magic.dm:511]:
```suggestion
			human_user.adjustBruteLoss(-10, FALSE)
			human_user.adjustFireLoss(-10, FALSE)
			human_user.adjustStaminaLoss(-10, FALSE)
			human_user.adjustToxLoss(-10, FALSE)
			human_user.adjustOxyLoss(-10)
```
This is a review I did multiple times before, and I'd like for you to understand. See, most procs that do damage also take an argument regarding updating health or not.
Updating health means recalculating all the damages, stat, redrawing the huds and whatnot.
If you are adding several kinds of damages, try to only update health at the end of it. There's no need to update it 5 times here, is it?

Comment (EdgeLordExe) [reply code/modules/antagonists/eldritch_cult/eldritch_magic.dm:511]: ah alright i never understood why you do the ,FALSE at the end

File: code/modules/antagonists/eldritch_cult/eldritch_magic.dm [hunk 3]
```diff
@@ -474,3 +474,39 @@
...
+			if(target.InCritical())
+				target.death()
+			target.adjustFireLoss(20)
+			new /obj/effect/temp_visual/eldritch_smoke(target.drop_location())
+			human_user.ExtinguishMob()
+			human_user.adjustBruteLoss(-10, FALSE)
+			human_user.adjustFireLoss(-10, FALSE)
+			human_user.adjustStaminaLoss(-10, FALSE)
+			human_user.adjustToxLoss(-10, FALSE)
+			human_user.adjustOxyLoss(-10)
```
Comment (Rohesie) [inline code/modules/antagonists/eldritch_cult/eldritch_magic.dm:512]: Indentation

**Pull request merged**